Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/transform/AuditTransform.java

```diff
@@ -0,0 +1,579 @@
+
+    /**
+     *
+     * @param participantObjectCode
+     * @param participantObjectCodeRole
+     * @param participantObjectIdCode
+     * @param particpantObjectIdCodeSystem
+     * @param participantObjectIdDisplayName
+     * @return
+     */
+    protected ParticipantObjectIdentificationType createParticipantObjectIdentification(short participantObjectCode, short participantObjectCodeRole,
+        String participantObjectIdCode, String particpantObjectIdCodeSystem, String participantObjectIdDisplayName) {
+        ParticipantObjectIdentificationType participantObject = new ParticipantObjectIdentificationType();
+
+        // Set the Participation Object Typecode
+        participantObject.setParticipantObjectTypeCode(participantObjectCode);
+
+        // Set the Participation Object Typecode Role
+        participantObject.setParticipantObjectTypeCodeRole(participantObjectCodeRole);
+
+        // Set the Participation Object Id Type code
+        CodedValueType partObjIdTypeCode = new CodedValueType();
+        partObjIdTypeCode.setCode(participantObjectIdCode);
+        partObjIdTypeCode.setCodeSystemName(particpantObjectIdCodeSystem);
+        partObjIdTypeCode.setDisplayName(participantObjectIdDisplayName);
+        participantObject.setParticipantObjectIDTypeCode(partObjIdTypeCode);
+        return participantObject;
+    }
+
+    /**
+     * Create the <code>EventIdentificationType</code> for an audit log record.
+     *
+     * @param actionCode
+     * @param eventOutcome
+     * @param eventId
+     * @return <code>EventIdentificationType</code>
+     */
+    private static EventIdentificationType createEventIdentification(String actionCode, Integer eventOutcome,
+        CodedValueType eventId) {
+        EventIdentificationType eventIdentification = new EventIdentificationType();
+
+        // Set the Event Action Code
+        eventIdentification.setEventActionCode(actionCode);
+
+        // Set the Event Action Time
+        try {
+            java.util.GregorianCalendar today = new java.util.GregorianCalendar(TimeZone.getTimeZone("GMT"));
+            javax.xml.datatype.DatatypeFactory factory = javax.xml.datatype.DatatypeFactory.newInstance();
+            javax.xml.datatype.XMLGregorianCalendar calendar = factory.newXMLGregorianCalendar(
+                today.get(java.util.GregorianCalendar.YEAR), today.get(java.util.GregorianCalendar.MONTH) + 1,
+                today.get(java.util.GregorianCalendar.DAY_OF_MONTH),
+                today.get(java.util.GregorianCalendar.HOUR_OF_DAY), today.get(java.util.GregorianCalendar.MINUTE),
+                today.get(java.util.GregorianCalendar.SECOND), today.get(java.util.GregorianCalendar.MILLISECOND),
+                0);
+            eventIdentification.setEventDateTime(calendar);
+        } catch (DatatypeConfigurationException e) {
+            LOG.error("DatatypeConfigurationException when creating XMLGregorian Date");
+            LOG.error(" message: " + e.getMessage());
```

Comment:
Anywhere you log an exception, using the following format: LOG.error("<message>" + e.getLocalizedMessage(), e);

For example, in this case, try:

LOG.error("Exception when creating XMLGregorian Date" + e.getLocalizedMessage(), e);

Note that you don't need to explicitly list "DatatypeConfigurationException"; the stack trace will take care of that.